src/charm.py

```diff
@@ -1,104 +1,75 @@
 #!/usr/bin/env python3
-# Copyright 2021 Canonical Ltd.
+# Copyright 2022 Canonical Ltd.
 # See LICENSE file for licensing details.
-#
-# Learn more at: https://juju.is/docs/sdk
-
-"""Charm the service.
-
-Refer to the following post for a quick-start guide that will help you
-develop a new k8s charm using the Operator Framework:
-
-    https://discourse.charmhub.io/t/4208
-"""
+"""Deploy and manage the Cinder CSI plugin for K8s on OpenStack."""
 
 import logging
-
+from pathlib import Path
+
+from charms.openstack_cloud_controller_operator.v0.cloud_config import (
+    CloudConfigRequires,
+)
+from charms.openstack_cloud_controller_operator.v0.lightkube_helpers import (
+    LightKubeHelpers,
+)
+from lightkube.resources.storage_v1 import StorageClass
 from ops.charm import CharmBase
-from ops.framework import StoredState
 from ops.main import main
-from ops.model import ActiveStatus
+from ops.model import ActiveStatus, BlockedStatus, MaintenanceStatus, WaitingStatus
 
 logger = logging.getLogger(__name__)
 
 
-class OperatorTemplateCharm(CharmBase):
-    """Charm the service."""
+class CinderCSIOperatorCharm(CharmBase):
+    """Deploy and manage the Cinder CSI plugin for K8s on OpenStack."""
 
-    _stored = StoredState()
+    manifests = Path("upstream/manifests")
+    version = Path("upstream/version").read_text()
```

Comment:
what's the relative path?  Is this relative to the charm agent directory?
Goes to look at `charmcraft.yaml` but doesn't see specific inclusion of anything.  

would you mind running a `charmcraft pack` and showing me those files are in the charm somewhere?

Reply:
Ok, fixed:

```
$ unzip -l cinder-csi-operator_ubuntu-20.04-amd64.charm | grep upstream
     2317  2022-02-14 20:17   upstream/update
       19  2022-02-14 20:21   upstream/version
      609  2022-02-14 20:21   upstream/README.md
      194  2022-02-14 20:21   upstream/manifests/cinder-csi-plugin/csi-cinder-driver.yaml
      432  2022-02-14 20:21   upstream/manifests/cinder-csi-plugin/csi-secret-cinderplugin.yaml
     5296  2022-02-14 20:21   upstream/manifests/cinder-csi-plugin/cinder-csi-controllerplugin-rbac.yaml
      694  2022-02-14 20:21   upstream/manifests/cinder-csi-plugin/cinder-csi-nodeplugin-rbac.yaml
     3705  2022-02-14 20:21   upstream/manifests/cinder-csi-plugin/cinder-csi-nodeplugin.yaml
     4539  2022-02-14 20:21   upstream/manifests/cinder-csi-plugin/cinder-csi-controllerplugin.yaml
```

Reply:
🚀